src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -2176,6 +2179,49 @@ new_fxstat(int vers, int fd, struct stat *buf)
 	return 0;
 }
 
+int
+fstat(int fd, struct stat *buf)
+{
+	int rc, fd_directed;
+
+	if (next_fstat == NULL) {
+		next_fstat = dlsym(RTLD_NEXT, "fstat");
+		D_ASSERT(next_fstat != NULL);
+	}
+	if (!hook_enabled)
+		return next_fstat(fd, buf);
+
+	fd_directed = get_fd_redirected(fd);
+	if (fd_directed < FD_FILE_BASE)
+		return next_fstat(fd, buf);
+
+	if (fd_directed < FD_DIR_BASE) {
+		rc          = dfs_ostat(file_list[fd_directed - FD_FILE_BASE]->dfs_mt->dfs,
+					file_list[fd_directed - FD_FILE_BASE]->file, buf);
+		buf->st_ino = file_list[fd_directed - FD_FILE_BASE]->st_ino;
+	} else {
+		rc          = dfs_ostat(dir_list[fd_directed - FD_DIR_BASE]->dfs_mt->dfs,
+					dir_list[fd_directed - FD_DIR_BASE]->dir, buf);
+		buf->st_ino = dir_list[fd_directed - FD_DIR_BASE]->st_ino;
+	}
+
+	if (rc) {
+		D_ERROR("dfs_ostat() failed: %d (%s)\n", rc, strerror(rc));
```

Comment:
```suggestion
		DS_ERROR(rc, "dfs_ostat() failed");
```

Reply:
Changes made for all D_ERROR(). Thank you!